insonmnia/miner/network/tinc_driver.go

```diff
@@ -0,0 +1,392 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	cnet "github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-plugins-helpers/ipam"
+	"github.com/docker/go-plugins-helpers/network"
+	log "github.com/noxiouz/zapctx/ctxlog"
+	"github.com/pborman/uuid"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+func NewTincNetwork(ctx context.Context, client *client.Client, config *TincNetworkConfig) (*TincNetworkDriver, *TincIPAMDriver, error) {
```

Comment:
There's a type `TincNetwork`, and `NewTincNetwork()` is not a constructor for that type; we might want to change naming.